plugins/telemetry-plugin/src/telemetry-plugin.ts

```diff
@@ -21,6 +21,51 @@ export function start(context: theia.PluginContext) {
                 ['programming language', e.document.languageId]
             ]);
     });
+
+    let gitLogHandlerInitialized: boolean;
```

Comment:
hello, question: can't we hook on global command invocation instead of intercept the plugin's code ?

so we could just add `git.commit` and `git.push` as commands to intercept/send telemetry events ?

Reply:
I think it will be possible if we extend in some way Theia CommandRegistry, but not sure is it make sense.

Reply:
I think it would allow to plug more easily other new events (or to track for example all commands the user is invoking)
besides that, it'll be more robust than going inside internal structure/objects of some plug-ins.


It would be like just providing the name of the command to intercept and the value to insert in telemetry

```typescript
something.intercept('git.commit', 'COMMIT_LOCALLY');
something.intercept('git.push', 'PUSH_TO_REMOTE');
```

Reply:
@benoitf Do you think it could be kept for a next PR ?

Reply:
to me, it's looking as too fragile right now.

Reply:
in case 
`something.intercept('git.commit', 'COMMIT_LOCALLY');` how we will send parameters if needed?

Reply:
I noticed that there were no parameters right now, but you can opt for another way like

format is open :-)
```
something.intercep('git.commit', (callback, args, etc) => callback.send('COMMIT_LOCALLY', args[2].substring(123).toUpperCase().whatever()) )
```

Reply:
@benoitf  Other possible solution 
https://github.com/eclipse-theia/theia/commit/b39a852192256e333ed6f4e0a13bd48d127123ae#diff-ab6e2d4c57658cf60e550cfe9ba87eacR295
WDYT?

Reply:
@vparfonov I'm not sure to land on the correct link ? it's about activation events ?

Reply:
but event contains only the commandID, correct, not the args you were asking for ?

Reply:
Yep probably this link better https://github.com/eclipse-theia/theia/blob/master/packages/core/src/common/command.ts#L273 
We can add parameters to the event.
Because `intercept` in this case something new for me I don't know how it works

Reply:
ok, for me, intercept was a method you were implementing

basically, it waits that command is executed (for example tracking all events you mentioned) and then it send the telemetry event with associated name if there is a matching reference (git.commit --> COMMIT_LOCALLY)

Reply:
@benoitf Do you have an estimate of how long it would be to do the change you're discussing with @vparfonov ? If it's not more than a few hours, let's do the change in this PR and avoid adding some more technical debt.

Reply:
I don't think it's long task as the event is there (Vitalii showed a link) so it's just handle the command event callback and if it's matching command, send the expected telemetry event

Reply:
If I understand correctly, the event is on `theia` side and there is no API to track the event on `plugin` side. So we need to provide this ability first.
please correct me if I'm wrong.

Reply:
on plugin side we have `che.telemetry` object and implementation of that is not on plug-in side but on `ext` so it can hook on events.

this is why I proposed
```
che.telemetry.addIntercept('git.commit', 'COMMIT_LOCALLY')
```
or
```
che.telemetry.addIntercept('git.commit', (command) => {che.telemetry.event('COMMIT_LOCALLY'....)
```

Reply:
@vparfonov according to the last answers @benoitf provided, it seems to me that we should probably do that in the current PR.
cc @ibuziuk